addons/hr_expense/models/hr_expense.py

```diff
@@ -169,7 +169,7 @@ def _compute_state(self):
     @api.depends('quantity', 'unit_amount', 'tax_ids', 'currency_id')
     def _compute_amount(self):
         for expense in self:
-            if expense.product_id and not expense.product_has_cost:
+            if expense.product_id and not expense.product_has_cost and not expense.unit_amount:
```

Comment:
```suggestion
            if expense.product_id and not expense.unit_amount:
```

Both are computed from `expense.product_id.standard_price`, the difference is that one is editable, the other not. Checking the non editable one doesn't make sense according to your fix.